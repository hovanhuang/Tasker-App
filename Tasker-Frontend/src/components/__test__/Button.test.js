import React from "react";
import ReactDOM from "react-dom";
import Button from "../Button";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';


afterEach(cleanup)
it(" renders no crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
})

it(" render button correctly", () => {
    const { getByTestId } = render(<Button text = "Close"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Close");
})

it("matches snapshot", () => {
    const tree = renderer.create(<Button text = "Close"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})