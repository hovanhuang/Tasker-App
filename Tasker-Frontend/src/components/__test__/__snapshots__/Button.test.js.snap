// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="btn"
  data-testid="button"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  Close
</button>
`;
